Replace Remaining GSA/data URLs with their cisagov/dotgov-data Counterparts

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This updates the remaining URLs that reference [GSA/data](https://github.com/GSA/data) to instead use [cisagov/dotgov-data](https://github.com/cisagov/dotgov-data) versions.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

The [cisagov/dotgov-data](https://github.com/cisagov/dotgov-data) repository has been updated to include all of the files necessary for the `gather-domains.sh` script.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##
Verified I was pulling the correct URLs to replace. TravisCI testing is no longer viable so automated testing is out.
<!-- How did you test your changes? How could someone else test this PR? -->

<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [ ] All new and existing tests pass.

jsf9k reviewed (approved):
Thanks for taking care of this!

dav3r reviewed (approved):
🤽

mcdonnnj committed:
Replace GSA/data URLs with cisagov/dotgov-data versions

All of the files used in the gather-domains.sh script have been migrated from
the GSA/data repository to the cisagov/dotgov-data repository now.

mcdonnnj committed:
Bumping version from 1.4.1 to 1.4.2

jsf9k reviewed (approved):
Great!
